### ISSUE TITLE
[RH] 4 - Library page - filter [simple]

### ISSUE THREAD

**Issue by tbrlpld** [mozilla/foundation.mozilla.org#8313]
<img width="488" alt="image" src="https://user-images.githubusercontent.com/24797493/156824877-157fe1d0-f16d-4864-8a44-3f64d0e97154.png">

Enable filtering of search results based on: topic, date, author and region.

Don’t add the “filter value search” yet. The functionality to reduce the number of filter values for each filter category should not be added yet. That is for later.

Don’t bother with the count preview as well. That might get complex. [See this discussion](https://www.figma.com/file/6BXnlCvPcQrtIaxunvuBYc?node-id=21:275#156801380)

See also:
- [Research hub models](https://docs.google.com/document/d/19WoZxJvisMsHnXqA4GDX3Te6t9JnqAI2NG5jE9J4LP4/edit#)
- [Designs](https://www.figma.com/file/6BXnlCvPcQrtIaxunvuBYc/Mozilla-Research%3A-Inspiration)

# Acceptance criteria

- [ ] Listed detail pages can be filtered by: topic, date, author and region.
- [ ] Per filter category show a list of all possible values and checkboxes to select the value.
- [ ] When arriving on a filtered page (e.g. through a link) then the filter from the URL should be displayed by a checked value.
- [ ] Multiple values per filter category can be selected. 
- [ ] Multiple filters (within the same and across different filter categories) are ANDed together (this means selecting more and more filters displayed less and less items, which feels natural)
- [ ] Previously active search filter remains active when other filters are added (and vice versa)
- [ ] The design assumes automatic reload of results, but we can have an "apply filters" button in desktop too. Only once that button is clicked are the results updated (with a full page reload). 
- [ ] When a filter value is selected and the "apply filter" button clicked, the page is reloaded and only detail pages which fulfill that filter value are displayed in the results list.

**Comment by bheasman** [comment]
@beccaklam please could you review this one from a design perspective? Thanks!

**Comment by beccaklam** [comment]
Hi @tbrlpld Can you provide a link to the PR for this work? Or if I can find the name of PR I can find it in the mofo-ra-foundation channel. Thanks!

**Comment by tbrlpld** [comment]
@beccaklam I am not yet working on this issue. I think the reviews are meant for the issue descriptions. This is at least what I though @Pomax meant. 

Maybe, @Pomax you can clarify. Did you mean the design team should review the tickets themselves or the implementation that is done based on the tickets (as part of the QA step)?

**Comment by beccaklam** [comment]
@tbrlpld Oh I see now, sorry about the misunderstanding! Got it, so I am doing a ticket review not a PR review.

**Comment by beccaklam** [comment]
@tbrlpld This looks fine to me, I'm not sure I'm the best to review what functionality we need for an MVP but what you have written feels right. The only question I have is regarding your last checkbox – loading results, it doesn't seem like a decision has been made. This would be good to bring up with Kasia. Personally I prefer auto reload but we would need to make sure there is good feedback (load animation?) for the user to let them know the filters have been applied which we don't have a design for.

**Comment by tbrlpld** [comment]
@beccaklam I have defined all dynamic behaviour as "nice to have". This means that instead of auto-updates the page needs to be refreshed. This makes it quite simple to implement while delivering the full functionality. We may loose a bit of "smoothness" of the experience, but it will work and is solid. The "auto-update" functionality can be added later as a progressive enhancement of the existing functionality. 

Same goes for pagination btw, which is the simper version of "load more".

**Comment by beccaklam** [comment]
Sounds good, will be good to bring up with Kasia to let her know that this is the decision for the MVP and that enhancements can be added in further sprints time permitting.

**Comment by bheasman** [comment]
Thanks both - agree with the initial approach here @tbrlpld -  but have added to the list for today to bring up with Kasia!
